packages/amplify-graphql-searchable-transformer/src/cdk/create-searchable-domain.ts

```diff
@@ -29,7 +28,7 @@ export const createSearchableDomain = (
     zoneAwareness: {
       enabled: false,
     },
-    domainName: Fn.conditionIf(HasEnvironmentParameter, Fn.ref('AWS::NoValue'), `d${apiId}`).toString(),
+    domainName: context.resourceHelper.generateDomainName(),
```

Comment:
Changing the domain name may be destructive for the existing apps. We need to verify that this change result in loss of data.